app-drawer/app-drawer.html

```diff
@@ -260,70 +260,139 @@
-              // very last track event may have zero ddx.
-              if (event.detail.ddx !== 0) {
-                this._lastTrackDirection = event.detail.ddx;
-              }
+              this._track(event);
               break;
-
             case 'end':
-              this.$.contentContainer.style.transitionDuration = '';
-              this.$.scrim.style.transitionDuration = '';
-              this.$.scrim.style.visibility = '';
-              this.$.scrim.style.opacity = '';
-              this.transform('', this.$.contentContainer);
-
-              // Only toggle the drawer if the user intentionally swipes in one direction.
-              if (Math.abs(event.detail.dx) > 30) {
-                this.opened = (this.position === 'left' && this._lastTrackDirection > 0) ||
-                  (this.position === 'right' && this._lastTrackDirection < 0);
-              }
+              this._trackEnd(event);
               break;
           }
         }
       },
 
+      _trackStart: function(event) {
+        // Disable transitions since style attributes will reflect user track events.
+        this.$.contentContainer.style.transitionDuration = '0s';
+        this.$.scrim.style.transitionDuration = '0s';
+        this.$.scrim.style.visibility = 'visible';
+
+        if (this.opened) {
+          this._translateOffset = 0;
+        } else if (this.position == 'left') {
+          this._translateOffset = -this.getWidth();
+        } else if (this.position == 'right') {
+          this._translateOffset = this.getWidth();
+        }
+
+        this._trackEvents = [];
+      },
+
+      _track: function(event) {
+        this._translateDrawer(event.detail.dx + this._translateOffset);
+        this._trackEvents.push(event);
+      },
+
+      _trackEnd: function(event) {
+        // Save translate progress made during track events for this._rafAnimate.
+        this._translateOffset = event.detail.dx + this._translateOffset;
+        this._startTimeStamp = performance.now();
+
+        // Track handler takes precedence over scrim tap handler.
+        this.cancelDebouncer('_scrimTapHandler');
+
+        // Calculate the velocity with the first track event that is recent enough, if
+        // it exists
+        this._velocity = 0;
+        var timeLowerBound = Date.now() - 100;
+        for (var i = 0; i < this._trackEvents.length; ++i) {
+          var trackEvent = this._trackEvents[i];
+          if (trackEvent.timeStamp >= timeLowerBound) {
+            var dx = event.detail.dx - trackEvent.detail.dx;
+            var dt = event.timeStamp - trackEvent.timeStamp;
+            this._velocity = dx / dt;
+            break;
+          }
+        }
+
+        if (this._velocity < -0.5 || this._velocity > 0.5) {
+          // If velocity is above the threshold, animate the drawer with the same velocity.
+          requestAnimationFrame(this._rafAnimate.bind(this));
```

Comment:
Would it be possible to try to compute the bezier curve http://cubic-bezier.com/#.14,.16,.31,.8 and the time of the transition for the transform? It doesn't have to be super accurate, but it seems like it'd be worth experimenting if you have time.